neat/tools/fittools/compartmentfitter.py

```diff
@@ -850,23 +861,112 @@ def plotSOV(self, alphas, phimat, importance, n_mode=8, alphas2=None):
         ax3.set_ylabel(r'$\phi_k(x_i)$')
         ax3.legend(loc=0)
 
-    def _getKernels(self, a, c):
+    def _constructKernels(self, a, c):
         nn = len(self.tree.getLocs('fit locs'))
         return [[Kernel((a, c[:,ii,jj])) for ii in range(nn)] for jj in range(nn)]
 
-    def plotKernels(self, alphas, phimat, t_arr=None):
-        fit_locs = self.tree.getLocs('fit locs')
-        nn = len(fit_locs)
+    def _getKernels(self, alphas=None, phimat=None,
+                          recompute=False, pprint=False):
+        """
```

Comment:
this docstring seems to be the same as for `getKernels`. why do you need both functions? they only seem to differ in their arguments

Reply:
`getKernels` provides a cleaner interface, and is part of the API. `_getKernels` adds a functionality, namely that optionally the SOV matrices alphas and phimat can be provided, which avoids recomputing them in some internal functions. End users however would have little need for this.